### PR TITLE
For #28769, update desktop view so template project's do not show up.

### PR DESCRIPTION
**Pull request by jfboismenu-adsk**
Reimplemented #28769 since it broke on pre-6.0 sites. We're now validating that the field exists before trying to filter on it.

### CODE REVIEW

File: python/tk_desktop/project_model.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 shotgun_model = sgtk.platform.import_framework("tk-framework-shotgunutils", "shotgun_model")
 shotgun_login = sgtk.platform.import_framework("tk-framework-login", "shotgun_login")
+shotgun_api3 = sgtk.platform.import_framework("tk-framework-login", "shotgun_api3")
```
Comment (robblau) [inline python/tk_desktop/project_model.py]: This seems like it should be

```python
from tank_vendor import shotgun_api3
```

Simpler and we're getting out of the business of the login framework handling connections, which this feels related to.

File: python/tk_desktop/project_model.py [hunk 2]
```diff
@@ -214,6 +215,22 @@ class SgProjectModel(ShotgunModel):
     thumbnail_updated = QtCore.Signal(QtGui.QStandardItem)
     project_launched = QtCore.Signal()
 
+    _supports_project_templates = None
+
+    @staticmethod
```
Comment (robblau) [inline python/tk_desktop/project_model.py]: Not sure if the difference is intentional (to lock subclasses from being able to change this behavior) but I'd expect this to be

```python
@classmethod
def supports_project_templates(cls, connection):
...
```

And then replace the ```SgProjectModel``` explicit symbols in the definition with ```cls```.
Is there a reason to make it static as opposed to class?

File: python/tk_desktop/project_model.py [hunk 3]
```diff
@@ -214,6 +215,22 @@ class SgProjectModel(ShotgunModel):
...
+    def supports_project_templates(connection):
+        # If we haven't checked on the server yet.
```
Comment (robblau) [inline python/tk_desktop/project_model.py]: Doc string?  We should include the reason why we are doing this check (Template projects in 6.0) so whoever looks at this 5 years from now understands the motivation for the check.

File: python/tk_desktop/project_model.py [hunk 4]
```diff
@@ -232,6 +249,11 @@ def __init__(self, parent, overlay_parent_widget):
             ["archived", "is_not", True],
             ["is_template", "is_not", True]
```
Comment (robblau) [inline python/tk_desktop/project_model.py]: ```is_template``` should be removed from the default list, no?

File: python/tk_desktop/project_model.py (hunk 1 above)
Comment (jfboismenu-adsk) [inline python/tk_desktop/project_model.py]: That's what I did at first. Unfortunately, the exception being thrown is from tk-framework-login's version of shotgun_api3. Therefore, if I catch the Fault exception from tank_vendor.shotgun_api3, it passes right through... That's why I am stuck catching the one from tk-framework-login.shotgun_api3.

File: python/tk_desktop/project_model.py (hunk 2 above)
Comment (jfboismenu-adsk) [inline python/tk_desktop/project_model.py]: Sorry, old habits die hard. I only started using @classmethod recently and therefore often forget about it. The code looks a lot better using @classmethod now. Thanks.

File: python/tk_desktop/project_model.py (hunk 4 above)
Comment (jfboismenu-adsk) [inline python/tk_desktop/project_model.py]: Forgot to commit my file before pushing. :(

**Pull request closed without merging**